lib/db/collections_db.dart

```diff
@@ -36,28 +47,56 @@ class CollectionsDB {
   _initDatabase() async {
     Directory documentsDirectory = await getApplicationDocumentsDirectory();
     String path = join(documentsDirectory.path, _databaseName);
-    return await openDatabase(
-      path,
-      version: _databaseVersion,
-      onCreate: _onCreate,
-    );
+    return await openDatabaseWithMigration(path, dbConfig);
+  }
+
+  static List<String> onCreate(String tableName) {
+    return [
+      '''
+				CREATE TABLE $tableName (
+					$columnID INTEGER PRIMARY KEY NOT NULL,
+					$columnOwner TEXT NOT NULL,
+					$columnEncryptedKey TEXT NOT NULL,
+					$columnKeyDecryptionNonce TEXT,
+					$columnName TEXT,
+					$columnType TEXT NOT NULL,
+					$columnEncryptedPath TEXT,
+					$columnPathDecryptionNonce TEXT,
+					$columnSharees TEXT,
+					$columnUpdationTime TEXT NOT NULL
+				);
+		'''
+    ];
+  }
+
+  static List<String> alterNameToAllowNULL() {
+    return [
+      ...onCreate(collectionsTableCopy),
+      '''
+				INSERT INTO $collectionsTableCopy
+				SELECT *
+				FROM $collectionsTable;
+      ''',
+      '''
+				DROP TABLE $collectionsTable;
+      ''',
+      '''
+				ALTER TABLE $collectionsTableCopy 
+				RENAME TO $collectionsTable;
+    '''
+    ];
   }
 
-  Future _onCreate(Database db, int version) async {
-    await db.execute('''
-                CREATE TABLE $collectionsTable (
-                  $columnID INTEGER PRIMARY KEY NOT NULL,
-                  $columnOwner TEXT NOT NULL,
-                  $columnEncryptedKey TEXT NOT NULL,
-                  $columnKeyDecryptionNonce TEXT,
-                  $columnName TEXT NOT NULL,
-                  $columnType TEXT NOT NULL,
-                  $columnEncryptedPath TEXT,
-                  $columnPathDecryptionNonce TEXT,
-                  $columnSharees TEXT,
-                  $columnUpdationTime TEXT NOT NULL
-                )
-                ''');
+  static List<String> addEncryptedName() {
+    return [
+      '''
+				ALTER TABLE $collectionsTable
+				ADD COLUMN $columnEncryptedName TEXT;
+      ''',
+      '''ALTER TABLE $collectionsTable
```

Comment:
Maintain consistency in formatting?

Reply:
![image](https://user-images.githubusercontent.com/46242073/105622456-f6195680-5e37-11eb-9e59-2829a12029a0.png)

Locally the file looks like this 😕